catching an error within a script

So, I was using the latest version of Eval Filter and I was attempting to process some JSON data. The data should be consistent in terms of the fields that it contains, but in some cases, they are not. So, this simple script, 

```
counter = 0;
if foo.bar == true {
    counter++;
}
```
results in the following panic if foo or foo.bar is not present

```
panic: reflect: call of reflect.Value.Interface on zero Value
```

Is there a way we could support a try/catch? This would allow the system to raise an error, and then the caller can choose what to do with the error by handling it or ignoring it depending on their use case.

--- a/evalfilter_test.go
+++ b/evalfilter_test.go
@@ -1487,3 +1487,34 @@ func TestRace(t *testing.T) {
 
 	wg.Wait()
 }
+
+// Test we can handle null values set in JSON objects
+func TestNullJsonField(t *testing.T) {
+	input := []string{
+		`{"event_count": null, "source": "************"}`,
+		`{"event_count": "332", "source": "************"}`,
+	}
+
+	obj := New(`if ( source == "************") { return true; } return false;`)
+
+	for _, line := range input {
+
+		err := obj.Prepare()
+		if err != nil {
+			t.Fatalf("error preparing")
+		}
+
+		// decode the line into JSON
+		jsonMap := make(map[string]interface{})
+		jsonErr := json.Unmarshal([]byte(line), &jsonMap)
+		if jsonErr != nil {
+			t.Errorf("error decoding json")
+		}
+
+		_, err = obj.Run(jsonMap)
+		if err != nil {
+			t.Errorf("Found unexpected error running test %s\n",
+				err.Error())
+		}
+	}
+}

--- a/vm/vm.go
+++ b/vm/vm.go
@@ -940,6 +940,13 @@ func (vm *VM) primitiveToObject(field reflect.Value) object.Object {
 	//
 	timeKind := reflect.TypeOf(time.Time{}).Kind()
 
+	//
+	// Invalid value?  Return null
+	//
+	if !field.IsValid() {
+		return &object.Null{}
+	}
+
 	switch field.Kind() {
 
 	case reflect.Map:

